src/hd/pci.c

```diff
@@ -1752,5 +1760,161 @@ void hd_read_uisvirtpci(hd_data_t *hd_data)
   free_str_list(sf_bus);
 }
 
+
+/*
+ * mmc
+ */
+void hd_read_mmc(hd_data_t *hd_data)
+{
+  int cnt;
+  hd_t *hd;
+  str_list_t *sf_bus, *sf_bus_e;
+  char *sf_dev, *drv, *drv_name, *modalias, *mmc_type;
+
+  sf_bus = read_dir("/sys/bus/mmc/devices", 'l');
+
+  if(!sf_bus) {
+    ADD2LOG("sysfs: no such bus: mmc\n");
+    return;
+  }
+
+  for(sf_bus_e = sf_bus; sf_bus_e; sf_bus_e = sf_bus_e->next) {
+    sf_dev = new_str(hd_read_sysfs_link("/sys/bus/mmc/devices", sf_bus_e->str));
+
+    ADD2LOG(
+      "  mmc device: name = %s\n    path = %s\n",
+      sf_bus_e->str,
+      hd_sysfs_id(sf_dev)
+    );
+
+    drv_name = NULL;
+    drv = new_str(hd_read_sysfs_link(sf_dev, "driver"));
+    if(drv) {
+      drv_name = strrchr(drv, '/');
+      if(drv_name) drv_name++;
+    }
+
+    ADD2LOG("    driver = \"%s\"\n", drv_name);
+
+    if((modalias = get_sysfs_attr_by_path(sf_dev, "modalias"))) {
+      modalias = canon_str(modalias, strlen(modalias));
+      ADD2LOG("    modalias = \"%s\"\n", modalias);
+    }
+
+    if(!sf_bus_e->str || sscanf(sf_bus_e->str, "mmc%u", &cnt) != 1) {
+      cnt = -1;
+    }
+    ADD2LOG("    index = %d\n", cnt);
+
+    if((mmc_type = get_sysfs_attr_by_path(sf_dev, "type"))) {
+      mmc_type = canon_str(mmc_type, strlen(mmc_type));
+      ADD2LOG("    type = \"%s\"\n", mmc_type);
+    }
+
+    hd = add_hd_entry(hd_data, __LINE__, 0);
+    hd->vendor.id = MAKE_ID(TAG_SPECIAL, 0x6015);	/* mmc */
```

Comment:
Where comes this number from? Kernel? Libhd itself? How can I know it is correct? Please add a comment.